### PR TITLE
Move integration tests to their own crate

### PR DESCRIPTION
**Pull request by justinmoon**
Fixes circular dependency issues described in https://github.com/fedimint/minimint/issues/123

Not sure if we want to go this route but figured I'd try it

### CODE REVIEW

File: tests/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#[cfg(test)]
+mod fixtures;
+
+#[cfg(test)]
+mod tests;
```
Comment (justinmoon) [inline tests/src/lib.rs:5]: If I put all the tests in here the compiler would complain that all the imports were unused. If I put them in their own module and marked it as a test module (I think that's what this does) then it no longer complained. LMK if there's a better way to do this ...

Comment (justinmoon) [reply tests/src/lib.rs]: Added another commit where I moved the tests from `tests/src` to `tests/tests` so you don't need to do this `#[cfg(test)]` stuff. Think this is probably better.

**Comment by justinmoon** [comment]
There was a CI failure with `No wallet is loaded. Load a wallet using loadwallet or create a new one with createwallet. (Note: A default wallet is no longer automatically created)`. Forced push a whitespace change to re-run CI ...

**Comment by justinmoon** [comment]
Actually that' is very weird. Did we just upgrade the version of bitcoind we're using or something? We don't call `createwallet` at all in `integrationtest.sh` ... I suspect re-running won't help ... I figured there was some kind of race condition ...

**Review by elsirion — APPROVED** [review]
Looks good, I assume the tests are still run implicitly by `cargo test` so we don't need to change CI?

EDIT: taking a look at supposedly unrelated CI failure

**Comment by justinmoon** [comment]
The CI failure fixed in https://github.com/fedimint/minimint/pull/126

**Comment by jkitman** [comment]
> There was a CI failure with `No wallet is loaded. Load a wallet using loadwallet or create a new one with createwallet. (Note: A default wallet is no longer automatically created)`. Forced push a whitespace change to re-run CI ...

So the reason that you are seeing that issue now is because
`cargo test -p minimint -- --test-threads=1`
doesn't run the integration tests if you move them to a new module.  They were creating the wallet so having the `create wallet` command was redundant, although it doesn't hurt to have it there I suppose.

That command appears in both `scripts/integrationtest.sh` and `scripts/final-checks.sh` if you update it in both locations the tests will run again in those scripts.

**Comment by justinmoon** [comment]
Thanks @jkitman

Updated the `cargo run -p` commands in out shell scripts and also reverted https://github.com/fedimint/minimint/pull/126.

**Comment by jkitman** [comment]
nit: Maybe we should put the tests into `integrationtests/tests` rather than `tests/tests`

Other than that looks good!

**Review by jkitman — APPROVED** [review]

**Comment by justinmoon** [comment]
> nit: Maybe we should put the tests into `integrationtests/tests` rather than `tests/tests`
> 
> Other than that looks good!

Done

**Pull request merged**